contracts/cw-escrow/src/msg.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_std::Uint128;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::state::CheckedCounterparty;
+
+/// Information about the token being used on one side of the escrow.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum TokenInfo {
+    /// A native token.
+    Native { denom: String, amount: Uint128 },
+    /// A cw20 token.
+    Cw20 {
+        contract_addr: String,
+        amount: Uint128,
+    },
+}
+
+/// Information about a counterparty in this escrow transaction and
+/// their promised funds.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct Counterparty {
```

Comment:
stylistic point, but I prefer instead of custom TokenInfo using an amount field and the a denom field which is a type from the cw20 package.

Reply:
`cw20::Denom` is a nice type. My only issue with it is that it isn't safe to use in contract message types as the `Cw20` variant holds an `Addr` so either way we've got to roll our own type here for the message section which removes some of the advantage of being "familiar" to callers. of the contract. I also like with these types that you can name them `Checked..` to indicate that this is a verified version of the type.

Happy to swap it over though if that isn't convincing. To quote @elsehow this is a "strong opinion weakly held".